### PR TITLE
Fix: USSD cont'd (Enabled sources)

### PR DESCRIPTION
**Pull request by Ifycode**
This pull request makes the following changes:
- To be merged into #1808 based on [this comment](https://github.com/ushahidi/platform-client/pull/1808#discussion_r954686427).

Testing checklist:
- [x] Sources come from config file, and default is set to 'web', 'sms', 'twitter', 'email'. 
- [x] Show or hide ussd and whatsapp sources checkboxes in post toolbar dropdown depending on if they are part of ENABLED_SOURCES
- [x] I certify that I ran my checklist

Ping @ushahidi/platform

### CODE REVIEW

**Comment by Ifycode** [comment]
@tuxpiper please help take a look.

**Review by tuxpiper — CHANGES_REQUESTED** [review]

File: app/config.js.j2 [hunk 1]
```diff
@@ -15,5 +15,6 @@ window.ushahidi = {
   intercomAppId: '{{ INTERCOM_APPID }}',
   appStoreId: '{{ APP_STORE_ID }}',
   ravenUrl: '{{ RAVEN_URL }}',
-  tosReleaseDate: '{{ TOS_RELEASE_DATE }}'
+  tosReleaseDate: '{{ TOS_RELEASE_DATE }}',
+  enabledSources: '{{ ENABLED_SOURCES }}'
```
[the remark this answers is not in the document]
Comment (Ifycode) [reply app/config.js.j2:19]: > If the environment variable is not present, jinja2 will not add the entry to window.ushahidi, and app.js should use a default instead.

This is the part I needed to hear @tuxpiper Thank you so much

File: legacy/app/app.js [hunk 2]
```diff
@@ -119,7 +119,8 @@ angular
             ? new Date(window.ushahidi.tosReleaseDate)
             : false, // Date in UTC
         EXPORT_POLLING_INTERVAL:
-            window.ushahidi.export_polling_interval || 30000
+            window.ushahidi.export_polling_interval || 30000,
+        ENABLED_SOURCES: window.ushahidi.sources.filter(source => source !== 'ussd' && source !== 'whatsapp')
```
[the remark this answers is not in the document]
Comment (Ifycode) [reply legacy/app/app.js:123]: So by this you mean this line should have been this ?

````
ENABLED_SOURCES: window.ushahidi.sources || ['sms', 'twitter', 'web', 'email']

````

File: root/src/config.js [hunk 3]
```diff
@@ -5,4 +5,6 @@
 // 	backendUrl : "https://ushahidi-platform-api-release.herokuapp.com",
 // 	mapboxApiKey: ""
 // };
-window.ushahidi = {};
+window.ushahidi = {
+    sources: ['sms', 'twitter', 'web', 'email', 'whatsapp', 'ussd']
```
[the remark this answers is not in the document]
Comment (Ifycode) [reply root/src/config.js:9]: So from the other review comments, I think I see my mistake. It's that I never understood the purpose... 

The user should be the one to set it here config.js, that is if they want a change. So `window.ushahidi` object should stay empty in config.js.
But if the user doesn't add their own sources property in here, the default in the app.js would take effect: https://github.com/ushahidi/platform-client/pull/1825#discussion_r961541455

And then I could probably add to the commented out code in config.js to help the user know they can define the sources they want if they don't want default to take effect:
````
// sources: ['sms', 'twitter', 'web', 'email', 'whatsapp', 'ussd']
````

**Comment by Ifycode** [comment]
@tuxpiper I have made the new changes. Please help review. And also if I should merge into the main USSD PR.

**Comment by Ifycode** [comment]
@tuxpiper I have made the new changes and fixed failing tests. Please help review. And also if I should merge into the main USSD PR.

**Review by tuxpiper — APPROVED** [review]
Great job @Ifycode  🙏  It's ready to merge!

**Pull request merged**